neat-form-core/src/main/java/com/nerdstone/neatformcore/form/json/JsonFormBuilder.kt

```diff
@@ -94,13 +99,42 @@ class JsonFormBuilder() : FormBuilder {
     internal fun parseJsonForm() {
         form = when {
             this::formString.isInitialized && formString.isNotNull() -> parseJson<NForm>(formString)
-            fileSource.isNotNull() -> parseJson<NForm>(
-                AssetFile.readAssetFileAsString(context, fileSource!!)
-            )
+            fileSource.isNotNull() -> {
+
+                val bundleName = fileSource!!.substring(fileSource!!.lastIndexOf('/') + 1, fileSource!!.lastIndexOf('.'))
+                val rawJsonStringTemplate: String = AssetFile.readAssetFileAsString(context, fileSource!!)
+
+                val interpolatedLocale = getStringSubstitutor(bundleName, Locale.getDefault())?.replace(rawJsonStringTemplate)
+                val interpolatedDefaultLocale = getStringSubstitutor(bundleName, Locale.ENGLISH)?.replace(interpolatedLocale)
+                        ?: interpolatedLocale ?: rawJsonStringTemplate
+
+                interpolatedDefaultLocale?.let {
+                    parseJson<NForm>(
+
+                            it
+                    )
+                }
+
+            }
             else -> null
         }
     }
 
+    private fun getStringSubstitutor(basename: String, locale: Locale): StringSubstitutor? {
+        val key = basename + "_" + locale
+        var substitutor: StringSubstitutor? = substitutorMap.getOrDefault(key, null)
+
+        if (substitutor == null) {
+            substitutor = LanguageHelper.getBundleStringSubstitutor(basename, locale)
+            if (substitutor != null) {
+                substitutorMap[key] = substitutor
+            }
```

Comment:
I this null check necessary when the `substitutor` object has been set from the line above it?

Reply:
Yeah the `LanguageHelper.getBundleStringSubstitutor(basename, locale)` might result with a null (if no such resource for the locale is configured in the app)

Reply:
Will update the null handling to be more Kotliny 😄